OpenRA.Game/ObjectCreator.cs

```diff
@@ -54,7 +55,13 @@ public ObjectCreator(Manifest manifest, FileSystem.FileSystem modFiles)
 				Assembly assembly;
 				if (!ResolvedAssemblies.TryGetValue(hash, out assembly))
 				{
-					assembly = Assembly.Load(data);
+					Stream debugStream = null;
+					if (Type.GetType("Mono.Runtime") != null && modFiles.TryOpen(path + ".mdb", out debugStream))
+						assembly = Assembly.Load(data, debugStream.ReadAllBytes());
+					else if (modFiles.TryOpen(path.Substring(0, path.Length - 4) + ".pdb", out debugStream))
```

Comment:
This line should be:
```C#
Type.GetType("Mono.Runtime") == null && modFiles.TryOpen(path + ".mdb", out debugStream)
```
Otherwise, if `modFiles.TryOpen(path + ".mdb", out debugStream)` returns false and `modFiles.TryOpen(path.Substring(0, path.Length - 4) + ".pdb", out debugStream)` return true, then `assembly = Assembly.Load(data, debugStream.ReadAllBytes());` will be executed.

Another possibility:
```C#
					Stream debugStream = null;
					var usingMono = Type.GetType("Mono.Runtime") != null;
					if (usingMono && modFiles.TryOpen(path + ".mdb", out debugStream))
						assembly = Assembly.Load(data, debugStream.ReadAllBytes());
					else if (!usingMono && modFiles.TryOpen(path.Substring(0, path.Length - 4) + ".pdb", out debugStream))
						assembly = Assembly.Load(data, debugStream.ReadAllBytes());
					else
						assembly = Assembly.Load(data);
```